src/platform/FreeBSD.cc

```diff
@@ -189,6 +219,19 @@ void FreeBSD::Execute(const CommandLine& c) const
 	}
 	setenv("LD_LIBRARY_PATH_FDS", libs.c_str(), 1);
 
+    int shm_FD = po_pack(map);
+
+    if(setenv("SHARED_MEMORYFD", std::to_string(shm_FD).c_str() ,1)){
+            err(-1, "SHARED_MEMORYFD not set");
+            }
+
+    fcntl(shm_FD, F_SETFD, 0);
+
+    auto lib_preopen = std::string("libpreopen.so");
```

Comment:
Why do you allocate a `std::string` as a named variable here, when you just used `std::to_string(shmfd).c_str()` above?